Wikipedia/Code/WMFArticleListTableViewController.m

```diff
@@ -259,4 +252,83 @@ - (void)tableViewUpdater:(WMFTableViewUpdater *)updater didUpdateTableView:(UITa
     });
 }
 
+#pragma mark - Sharing
+
+- (UIActivityViewController *)shareActivityController:(NSURL *)url {
+    WMFArticle *article = [self.userDataStore fetchArticleWithURL:url];
+    NSMutableArray *items = [NSMutableArray array];
+    
+    
+    // TODO: Use WMFArticleTextActivitySource
+    NSString *text =  [NSString stringWithFormat:@"\"%@\" on @Wikipedia", [article displayTitle]];
+    [items addObject:text];
+    
+    NSURL *desktopURL = [NSURL wmf_desktopURLForURL:url];
+    if (desktopURL) {
+        NSURLComponents *components = [NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:NO];
+        
+        NSURLQueryItem *queryItem = [NSURLQueryItem queryItemWithName:@"wprov" value:@"sfsi1"];
+        components.queryItems = @[queryItem];
+        
+        NSURL *componentsURL = components.URL;
+        if (componentsURL) {
+            [items addObject:componentsURL];
+        }
+    }
+    
+    MKMapItem *mapItem = [article mapItem];
+    if (mapItem) {
+        [items addObject:mapItem];
+    }
+
+    UIActivityViewController *vc = [[UIActivityViewController alloc] initWithActivityItems:items
+                                                                                              applicationActivities:
+                                                             @[[[TUSafariActivity alloc] init],
+                                                               [[WMFOpenInMapsActivity alloc] init],
+                                                               [[WMFGetDirectionsInMapsActivity alloc] init]]];
+    return vc;
+}
+
+- (void)shareArticle:(UIActivityViewController *)shareActivityController {
+    [self presentViewController:shareActivityController animated:YES completion:NULL];
+}
+
+#pragma mark - Row actions
+
+- (NSString *)deleteActionText {
+    return WMFLocalizedStringWithDefaultValue(@"article-delete", nil, nil, @"Delete", @"Text of the article list row action shown on swipe which deletes the article");
+}
+
+- (NSString *)shareActionText {
+    return WMFLocalizedStringWithDefaultValue(@"article-share", nil, nil, @"Share", @"Text of the article list row action shown on swipe which allows the user to choose the sharing option");
+}
+
+- (NSString *)saveActionText {
+    return WMFLocalizedStringWithDefaultValue(@"article-save", nil, nil, @"Save", @"Text of the article list row action shown on swipe which allows the user to save the article");
+}
+
+- (UITableViewRowAction *)shareAction:(NSIndexPath *)indexPath {
```

Comment:
Thanks, Joe! I added `rowActionWithStyle ` to the VC and set `reloadRowsAtIndexPaths ` in `WMFTableViewUpdater.m` to `UITableViewRowAnimationNone ` but the animation still looks the same for me (as in 'before')